packages/react-native-oursky/src/VerifyOTP.js

```diff
@@ -0,0 +1,213 @@
+// @flow
+import * as React from "react";
+import { TouchableOpacity, View, StyleSheet } from "react-native";
+
+import Text from "./Text";
+import type { Props as ExtraTextProps } from "./ExtraText";
+import ExtraText from "./ExtraText";
+import { ViewStyle, TextStyle } from "./styles";
+import CodeBox from "./CodeBox";
+import type { Props as CodeBoxProps } from "./CodeBox";
+import TextInput from "./TextInput";
+
+const defaultStyles = StyleSheet.create({
+  box: {
+    borderRadius: 24,
+    backgroundColor: "white",
+    shadowColor: "rgb(0, 0, 0)",
+    shadowOffset: { width: 0, height: 10 },
+    shadowOpacity: 0.2,
+    elevation: 15,
+    paddingHorizontal: 21,
+    paddingVertical: 31,
+  },
+  description: {
+    fontSize: 18,
+    lineHeight: 24,
+    color: "rgb(51, 51, 51)",
+    marginBottom: 23,
+  },
+  codeBoxContainer: {
+    flexDirection: "row",
+    marginHorizontal: -4.5,
+  },
+  error: {
+    color: "rgb(238, 0, 0)",
+    alignSelf: "center",
+    marginTop: 12,
+  },
+  resendContainer: {
+    marginTop: 33,
+    alignItems: "center",
+  },
+  resendText: {
+    fontSize: 13,
+    color: "rgb(170, 170, 170)",
+  },
+  hiddenTextInput: {
+    width: 0,
+    height: 0,
+    opacity: 0,
+  },
+});
+
+export type Props = ExtraTextProps & {
+  description?: React.Node,
+  resendText?: React.Node,
+  resending?: boolean, // disabled resend button
+
+  style?: ViewStyle,
+  descriptionStyle?: TextStyle,
+  codeBoxStyle?: ViewStyle,
+  codeBoxTextStyle?: TextStyle,
+  resendContainerStyle?: ViewStyle,
+  resendTextStyle?: TextStyle,
+
+  onEnterCode?: (code: string, clearCode: () => void) => void,
+  onPressResend?: (restartTimer: () => void) => void,
+};
+
+type State = {
+  value: string,
+  countDownSecond: number,
+};
+
+type TextInputRefProps = {
+  textInputRef: React$Ref<TextInput>,
+};
+
+class VerifyOTP extends React.PureComponent<Props & TextInputRefProps, State> {
+  state = {
+    value: "",
+    countDownSecond: 59,
+  };
+
+  timerId: IntervalID | null = null;
```

Comment:
A more performant approach will be we introduce a `Timer` component responsible for count down because this component is `big` and re-render every second seems not so good but that is a improvement that we can do in future